eng/build.ps1

```diff
@@ -1,35 +1,52 @@
 [CmdletBinding(PositionalBinding=$false)]
 Param(
-  [string] $ArchGroup,
-  [switch] $Release,
-  [Alias("framework")][string] $TargetGroup,
-  [Alias("os")][string] $OSGroup,
-  [Alias("allconfigurations")][switch] $BuildAllConfigurations,
-  [string] $RuntimeOS,
-  [switch] $OuterLoop,
-  [switch] $SkipTests,
-  [Alias("sync")][switch] $Restore,
-  [switch] $Clean,
-  [Alias("includetests")][switch] $BuildTests,
-  [switch] $Test,
-  [switch] $Coverage,
-  [switch] $InitTools,
-  [Parameter(ValueFromRemainingArguments=$true)][String[]]$ExtraArgs
+  [switch] $buildtests,
+  [string][Alias('c')]$configuration = "Debug",
+  [string] $framework,
+  [string] $os,
+  [switch] $allconfigurations,
+  [switch] $coverage,
+  [switch] $outerloop,
+  [string] $arch,
+  [switch] $help,
+  [Parameter(ValueFromRemainingArguments=$true)][String[]]$properties
 )
 
-$defaultargs = "-restore -build -warnaserror:0"
+function Print-Usage() {
+    Write-Host "Default if no arguments are passed in: -restore -build"
+    Write-Host ""
+    Write-Host "CoreFX specific settings:"
+    Write-Host "  -buildtests             Build test projects. Can be used as a target or as an option."
+    Write-Host "  -framework              The target group assemblies are built for."
+    Write-Host "  -os                     The operating system assemblies are built for."
+    Write-Host "  -allconfigurations      Build packages for all build configurations."
+    Write-Host "  -coverage               Collect code coverage when testing."
+    Write-Host "  -outerloop              Include tests which are marked as OuterLoop."
+    Write-Host "  -arch                   The architecture group."
+    Write-Host ""
+}
+
+if ($help -or (($properties -ne $null) -and ($properties.Contains("/help") -or $properties.Contains("/?")))) {
+  Print-Usage
+  Invoke-Expression "& `"$PSScriptRoot/common/build.ps1`" -help"
+  exit 0
+}
 
-$possibleDirToBuild = if($ExtraArgs.Length -gt 0) { $ExtraArgs[0]; } else { $null }
+if ($PSBoundParameters.Count -eq 0) {
+  $arguments = "-restore -build"
```

Comment:
I need to find a better way to default to -restore -build if no target is passed in. Currently the default args only work if the script is invoked without any arguments. There's probably no way to implement these default args without hardcoding and checking all other targets.

cc @safern